### DISCUSSION TITLE
Is there a way Remove or Overwrite a Modal event handler?

### DISCUSSION THREAD

**Discussion by mathieu-fournier** [iTwin/iTwinUI-react#516] [Q&A] [answered]
Hi o/, 

Is there a way to remove/overwrite this event handler found in [Modal.tsx ](https://github.com/iTwin/iTwinUI-react/blob/406c9b6827e98cf9c475d845d698d4fa6d4b113a/src/core/Modal/Modal.tsx#L186)? 
![image](https://user-images.githubusercontent.com/82905333/150193144-688ab30c-60f1-4aba-9f83-e35a955aa340.png)

Stoppping the event propagation prevents from starting a Selection Area.
Here's a [codeSandbox of the selectionArea behaviour](https://codesandbox.io/s/react-selectionarea-stopevent-pvgsn?file=/src/App.js). 

Thanks (:

**Comment by mayank99** [comment, marked as answer]
Hi @mathieu-fournier 👋

Could you produce a sandbox showing how you're using the selectionArea in Modal? That handler is meant to stop propagating clicks into the backdrop (parent) which has its own handler for closing the modal. If the selectionArea is inside the modal, it shouldn't be a problem.

**Reply by mathieu-fournier** [reply]
Hi @mayank99 o/,

I've made [a sandbox](https://codesandbox.io/s/itwinui-react-minimal-example-forked-frk4u) using the selectionArea in Modal. 

Thanks (:

**Reply by mayank99** [reply]
Thanks for the sandbox. It looks like the problem is happening because the lib you're using is adding a listener to `document`. https://github.com/Simonwep/selection/blob/v2/src/index.ts#L111

@iTwin/itwinui-react-1 We'll need to think about a good approach to solve this. Either through composition (separating `ModalBackdrop` and `ModalDialog` so the user has control over both) or through a prop to disable the auto-close behavior for backdrop clicks.

**Reply by mathieu-fournier** [reply]
Do you have any clue on a possible workaround I could use to fix this on my side ?

**Reply by mayank99** [reply]
I can't think of any workaround. I created a PR with quick fix though: #518

**Reply by mathieu-fournier** [reply]
The PR will do, Thanks a lot (:
